modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStore.java

```diff
@@ -3285,6 +3290,9 @@ private Integer findVarcharColumnLength(AttributeDescriptor att) {
 
                     if (sqlType == sqlTypes[i]) {
                         sqlTypeNames[i] = sqlTypeName;
+                        // Geos-7533
+                        // cache the sqlType which required going to database for sql types
+                        getSqlTypeToSqlTypeNameOverrides().put(sqlType, sqlTypeName);
```

Comment:
Nope, this is not all right. The getSqlTypeToSqlTypeNameOverrides is public with a specific semantic defined in its javadoc, what you are registering here is not an override any longer.
I suggest you create a separate map that acts as a cache instead. Also, if you plan to update it this way, make it a concurrent hash map, otherwise code in production might experience a concurrent modification exception.

Reply:
we already have two hashmaps inside the implementation

would it be acceptable that if this method is made a synchronized method, so that if unseen fields are being fetched from database from user1, user2 can wait for the only one time. 

For hash map, if I use the main hashamp instead of the hashmap behind getSqlTypeToSqlTypeNameOverrides() will that do?

Reply:
fixed